Formula/souffle.rb

```diff
@@ -13,19 +13,22 @@ class Souffle < Formula
     sha256 cellar: :any_skip_relocation, big_sur:       "54359b0756972ca99d438dd8cbd72a49dfe93b53ee3c1731e7b32931973b5bfc"
   end
 
-  depends_on "bison" => :build # Bison included in macOS is out of date.
+  depends_on "bison" => :build 
   depends_on "cmake" => :build
-  depends_on "mcpp" => :build
   depends_on "pkg-config" => :build
-  uses_from_macos "flex" => :build
-  uses_from_macos "libffi"
-  uses_from_macos "ncurses"
-  uses_from_macos "sqlite"
-  uses_from_macos "zlib"
```

Comment:
Why not use the system versions?

Reply:
Seems that the `Formula["xxx"]` calls below will return paths that does not exist for the `uses_from_macos` dependencies. Any advice?

Reply:
Upstreams build system should not depend on hardcoded library paths. No other CMake builds do as far as I know.

Reply:
These paths are not used by cmake. 
Souffle is a datalog compiler, which, when working in compiler mode, will first transpile the datalog source files into C++ source files, and launches a C++ compiler to compile its generated C++ source files to binary. These paths are passed to the C++ compiler at that stage.

Essentially, the content in `souffle-compile.py` is a configuration, which is generated during build time, and contains the include paths, library paths, etc.

I understand that the paths should not be hardcoded. I am seeking for advice for setting these paths correctly.